### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkins-infra/helpdesk/issues/4340

ref https://github.com/jenkins-infra/backend-extension-indexer/pull/95#issuecomment-2408559791

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]
Looks reasonable.

CC @MarkEWaite

**Comment by timja** [comment]
infra.ci fails for:

```

[2024-10-12T16:09:10.081Z] [INFO] Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/apache/maven/plugins/maven-release-plugin/3.0.1/maven-release-plugin-3.0.1.pom

[2024-10-12T16:09:10.081Z] [WARNING] The POM for org.apache.maven.plugins:maven-release-plugin:jar:3.0.1 is missing, no dependency information available

[2024-10-12T16:09:10.081Z] [INFO] Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/apache/maven/plugins/maven-release-plugin/3.0.1/maven-release-plugin-3.0.1.jar

[2024-10-12T16:09:10.081Z] [WARNING] Failed to retrieve plugin descriptor for org.apache.maven.plugins:maven-release-plugin:3.0.1: Plugin org.apache.maven.plugins:maven-release-plugin:3.0.1 or one of its dependencies could not be resolved:

[2024-10-12T16:09:10.081Z] 	Could not find artifact org.apache.maven.plugins:maven-release-plugin:jar:3.0.1 in repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/)

[2024-10-12T16:09:10.081Z] 

[2024-10-12T16:09:10.081Z] [INFO] Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/apache/maven/plugins/maven-metadata.xml

[2024-10-12T16:09:10.081Z] [INFO] Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/codehaus/mojo/maven-metadata.xml

[2024-10-12T16:09:10.081Z] [INFO] Downloaded from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/codehaus/mojo/maven-metadata.xml (255 B at 9.4 kB/s)

[2024-10-12T16:09:10.081Z] [INFO] ------------------------------------------------------------------------

[2024-10-12T16:09:10.081Z] [INFO] BUILD FAILURE

[2024-10-12T16:09:10.081Z] [INFO] ------------------------------------------------------------------------

[2024-10-12T16:09:10.081Z] [INFO] Total time:  1.108 s

[2024-10-12T16:09:10.081Z] [INFO] Finished at: 2024-10-12T16:09:09Z

[2024-10-12T16:09:10.081Z] [INFO] ------------------------------------------------------------------------

[2024-10-12T16:09:10.081Z] [ERROR] No plugin found for prefix 'dependency' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/home/jenkins/.m2/repository), repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/)] -> [Help 1]
```

not sure why it fails there but is fine on ci.jenkins.io

**Comment by timja** [comment]
Maybe something to do with this:
```
16:08:06  WARNING: artifact caching proxy is enabled but the provided 'ARTIFACT_CACHING_PROXY_SERVERID' setup on the agent is empty, will use repo.jenkins-ci.org.
```


```
16:09:10  [INFO] Downloading from repo.jenkins-ci.org: https://repo.jenkins-ci.org/public/org/apache/maven/plugins/maven-resources-plugin/3.3.1/maven-resources-plugin-3.3.1.pom
16:09:10  [WARNING] The POM for org.apache.maven.plugins:maven-resources-plugin:jar:3.3.1 is missing, no dependency information available
```

**Comment by daniel-beck** [comment]
> not sure why it fails there but is fine on ci.jenkins.io

I have the same locally on `master`. I suspect this project is broken since repo.jenkins-ci,org stopped mirroring repo1. ci.jenkins.io may have a custom settings.xml that takes precedence.

**Comment by timja** [comment]
it does yes

File: maven-settings.xml [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+    </profile>
+  </profiles>
   <mirrors>
     <mirror>
-      <id>repo.jenkins-ci.org</id>
-      <url>https://repo.jenkins-ci.org/public/</url>
-      <mirrorOf>*</mirrorOf>
+      <id>repo.j.o-mirror</id>
+      <mirrorOf>repo.jenkins-ci.org</mirrorOf>
+      <url>https://repo.jenkine-ci.org/public/</url>
```
Comment (lemeurherve) [inline maven-settings.xml]: Typo?

```suggestion
      <url>https://repo.jenkins-ci.org/public/</url>
```

**Pull request merged**